### ISSUE TITLE
Support for 1.2.x Opensearch versions

### ISSUE THREAD

**Issue by Willshield** [grafana/opensearch-datasource#29]
Grafana version: 8.3.3
DS version:1.1.2
Opensearch version: 1.2.0

There are new versions of Opensearch available:
https://opensearch.org/docs/latest/version-history/

The currently latest version of this plugin supports 1.0.x however there are already 1.2.x and 1.3.x versions. We need to integrate a 1.2.0 instance. The support for this enhancement / update would be really appreciated.

Thanks

Currently supported versions:
![image](https://user-images.githubusercontent.com/13013769/167842754-394a1027-4c47-440c-8c10-5af64a4acc89.png)

**Comment by R0bertB** [comment]
1.0 is almost 1 year old, I also have issue because of this.

**Comment by EdeHeltovics** [comment]
This is indeed an important lack of possibilities.

**Comment by loganmarchione** [comment]
Would also like to see this. [Graylog 4.3 now supports OpenSearch](https://docs.graylog.org/docs/installing-opensearch) 1.1, 1.2, and 1.3, but Grafana only supports 1.0. Because of this, I'm sticking to Grafana 8.5.6.

**Comment by klubi** [comment]
Yeah... would be nice to get latest OpenSearch versions supported.
Any chance for it?

**Comment by matyax** [comment]
We need to decide the next steps for this one @grafana/observability-logs

**Comment by weaversam8** [comment]
Do we know that the existing version of the plugin is wholly incompatible with versions greater than 1.0.x? Is there a breaking change to the API?

**Comment by loganmarchione** [comment]
Graylog 5.0 is in beta and is adding support for OpenSearch up to 2.3. It would be nice to have support for newer OpenSearch versions. 

https://www.graylog.org/post/announcing-graylog-5-0-beta-1/



Also, do the versions of this datasource (e.g., v1 vs v2) track versions of OpenSearch?

https://github.com/grafana/opensearch-datasource/releases

**Comment by jonny-wg2** [comment]
Status on this?

**Comment by gabor** [comment]
the current situation is not ideal, but as long as you choose the "opensearch 1.0.x" option, you can just use it with any opensearch version. the plugin does not validate the real version of the opensearch database.

we will improve this functionality in the future, but for now this can be used as a workaround.

**Comment by gabor** [comment]
notes: we currently see the following possible approaches:
1. change `opensearch 1.0.x` to `opensearch 1.x` and add the `opensearch 2.x`  (and also always add one for all supported opensearch major versions)
2. change `opensearch 1.0.x` to `opensearch`, and use some form of version-detection when it becomes necessary to differentiate (for example, if the next major version of opensearch is not compatible with the previous versions)

**Comment by loganmarchione** [comment]
Was using:
- `grafana/grafana-oss:8.5.20` Docker image
- OpenSearch Grafana plugin 1.2.4
- OpenSearch 1.3.8

Upgraded Grafana, which also upgraded the OpenSearch plugin. I had to change all of my visualizations because I was changing from the ElasticSearch datasource to OpenSearch.
- `grafana/grafana-oss:9.3.2` Docker image
- OpenSearch Grafana plugin 2.0.5
- OpenSearch 1.3.8 (this did not change)

Then, upgraded OpenSearch to 2.5.0. All of my visualizations still worked.
- `grafana/grafana-oss:9.3.2` Docker image (this did not change)
- OpenSearch Grafana plugin 2.0.5 (this did not change)
- OpenSearch 2.5.0


@gabor - I think you should go with option 2 and change `opensearch 1.0.x` to `opensearch`. The current wording is misleading, everyone in this thread thought that the OpenSearch datasource plugin didn't work with anything other than `1.0.x`.

**Comment by fridgepoet** [comment]
Addressed this in https://github.com/grafana/opensearch-datasource/pull/153
